src/AutoloadPlugin.php

```diff
@@ -0,0 +1,74 @@
+<?php declare(strict_types=1);
+
+namespace Maho\ComposerPlugin;
+
+use Composer\Composer;
+use Composer\EventDispatcher\EventSubscriberInterface;
+use Composer\IO\IOInterface;
+use Composer\Package\RootPackage;
+use Composer\Plugin\PluginInterface;
+use Composer\Script\Event;
+use Composer\Script\ScriptEvents;
+use Composer\Util\Filesystem;
+
+final class AutoloadPlugin implements PluginInterface, EventSubscriberInterface
```

Comment:
what's the difference between AutoloadPlugin and AutoloadRuntime? I'd describe then in the code because it's not clear to me. I don't super love that it is becoming so much bigger compared to the few lines that it was originally :-\

Reply:
the thing is that now I don't know what does what, which is precisely what I didn't like about the previous plugins for openmage. the modman plugin is perfect buecause the name explains everything. Now I feel lost. Maybe it's just a matter of documentation but to me semplicity is key for maintainability and possibility for people to work on the project

Reply:
I've added a bunch of phpdoc blocks. I should note that even before the autoload runtime and autoload plugin were separated, it's just that we only had one `MahoComposerPlugin.php` file that handled all events. It thought it better to split it up, but it did cause two similar and confusing filenames. Hopefully it's better now.

Also I agree the OM plugin is way too complex and hard to understand with factory classes, etc. But still:

```sh 
$ sloccount magento-composer-installer/src/
SLOC    Directory       SLOC-by-Language (Sorted)
2289    src             php=2289

$ sloccount maho-composer-plugin/src/
SLOC    Directory       SLOC-by-Language (Sorted)
489     src             php=489
```

We are 1/5 the size of the codebase and already do so much more.